### ISSUE TITLE
Make app environment immutable on application restart/reload by default

### ISSUE THREAD

**Issue by Unitech** [Unitech/pm2#2471]
When an app is started via CLI or via CONFIG file, now on restart the environment will stay the same as the beginning.

To be able to update the environment / application configuration you will now have to pass the new option `--update-env`:

CLI:

``` bash
# Set the DEBUG variable
$ DEBUG="*" pm2 start app.js
# This does not work
$ DEBUG="app:*" pm2 restart app.js
# This will update the environment
$ DEBUG="app:*" pm2 restart app.js --update-env
```

Config file:

``` bash
$ pm2 start config.yml
# Update fields of the config.yml
$ pm2 restart config.yml
```
#2268 How do you NOT watch in production?
#2469 Watch & Reload causes double output/start of script
#2461 pm2 start --no-vizion --no-pmx myapplication.json - vizion and pmx enabled
#2356 Watch feature starting more processes then expected
#2227 For cluster mode apps when we make gracefulReload all logs begin combines to one
#2441 Watch & reload resets to disabled bug

Possibly related:
#2427 Port is in use

**Comment by Unitech** [comment]
Fix is available:

```bash
$ npm install pm2@next -g
$ pm2 update
```

**Comment by Unitech** [comment]
Patch available on pm2@2.1.4 (main):

```bash
$ npm install pm2 -g
$ pm2 update
```
